packages/infra/lib/api-stack.ts

```diff
@@ -791,6 +801,44 @@ export class APIStack extends Stack {
     return cdaToVisualizationLambda;
   }
 
+  private setupCWDocContribution(ownProps: {
+    baseResource: apig.Resource;
+    lambdaLayers: lambda.ILayerVersion[];
+    alarmAction: SnsAction | undefined;
+    authorizer: apig.IAuthorizer;
+    oauthScopes: cognito.OAuthScope[];
+    envType: string;
+    bucket: s3.Bucket;
+  }): Lambda {
+    const { baseResource, lambdaLayers, alarmAction, authorizer, oauthScopes, envType, bucket } =
+      ownProps;
+
+    const cwLambda = createLambda({
+      stack: this,
+      name: "CommonWellDocContribution",
+      runtime: lambda.Runtime.NODEJS_18_X,
+      entry: "cw-doc-contribution",
+      layers: lambdaLayers,
+      alarmSnsAction: alarmAction,
+      envVars: {
+        ENV_TYPE: envType,
+        ...(bucket && {
+          MEDICAL_DOCUMENTS_BUCKET_NAME: bucket.bucketName,
+        }),
+      },
+    });
+
+    const cwResource = baseResource.addResource("commonwell");
+    cwResource.addMethod("GET", new apig.LambdaIntegration(cwLambda), {
+      authorizer: authorizer,
+      authorizationScopes: oauthScopes.map(s => s.scopeName),
+    });
+
+    bucket.grantReadWrite(cwLambda);
```

Comment:
Does it need write access b/c of pre-signed URL? If not, it'd be good to limit to read access.